### PR TITLE
Civi\Angular\ChangeSet - Relax debug-mode consistency check

### PR DESCRIPTION
**Pull request by totten**
Overview
--------

The Angular subsystem *sometimes* includes a consistency-check. The consistency check ensures that an HTML document is read/written in consistent format (i.e. `encode(decode($html)) === $html`). However, the check is not often run, and it sometimes produces errors on consistency issues which we don't care about.

The approach here is to relax the checks in `civicrm-core` and reproduce them in separate developer-oriented tooling.

Before
------

* If an Angular HTML partial is *not* altered (`hook_civicrm_alterAngular`), then the partial is *not* checked for consistency.
* If a site is in production/non-debug mode, then the partial is *not* checked for consistency.
* If a site is in debug mode, then the partial *is* checked for consistency.
* tldr: The consistency-check is executed very rarely.

After
-----

* The partial is not checked for consistency.
* A task is recorded on the [Afform roadmap](https://github.com/totten/afform/blob/master/docs/roadmap.md) to include better consistency checks in the `afform_auditor` extension.

### CODE REVIEW

**Comment by colemanw** [comment]
> The Angular subsystem *sometimes* includes a consistency-check.

Sounds like the consistency check is inconsistent. Maybe it needs a consistency check.
